### DISCUSSION TITLE
What would happen if Publii disappeared?

### DISCUSSION THREAD

**Discussion by wefer-01** [GetPublii/Publii#1564] [General]
I have a website made with Publii that I put a lot of effort and time into. I am satisfied with Publii and for what I need on this website, it is perfect for me. For my part, I will continue to use the program and make donations to help the project.

But I have one fear, which has held me back from making more Publii-based websites: what would I do if one day the Publii project disappeared? How could I continue to update and maintain the website? Let's imagine I can't even use an old version because it's not supported.

Or another case: let's imagine that it starts to give me an error when doing the rendering. And I don't get a solution, so I'm left unable to continue the web. It happened to me in the past, but luckily I was able to pull back up.

Ultimately, it would be good to think about what possibilities there are to pass the information published on Publii to plain HTML, for example, or other systems. Because taking the render result directly and trying to edit it is a drama as it is really complex with scripts etc.

I don't know if I finished explaining myself well. Thanks anyway!

**Comment by johnwoltman** [comment]
I understand where you're coming from. I've used a lot of website systems over the years and I've lucked out that they're all still going strong today.

I would be **very** sad if Publii died.  If the team disbanded I would hope that someone else would take up the project, but there's no guarantee.

Now, having said that, here's my personal experience.  Every time I've migrated my website to a new platform, I did it by writing some conversion code.

- Wrote PHP to convert e107 to TextPattern, then to Drupal
- Wrote Python to convert Drupal to Django
- Wrote even more Python to convert Django to Dokuwiki.
- Wrote some more Python to convert Dokuwiki to Publii.

The one thing these systems have in common is they store their information in some kind of database.  It might be a pain to convert certain things (like Publii's image galleries) into a future platform, but it will be _possible_.

I think one thing we can all do is chip in (as we're able) to financially support Publii.  I've bought at least one plugin which I'm not even currently using, I just want to the dev team to make some money.  I'd be very sad if Publii went away.  It's everything I've been searching for in a website editor since, no lie, like 2004.

**Reply by wefer-01** [reply]
I somehow wish Publii's design would take into account the possibility of this happening, or even have a tool to allow content to be exported to at least classic HTML for those people who want to switch to another system
I personally do not dare to take the step of moving other websites to Publii for fear of being "locked" in the future within this system, and for new websites I end up preferring standardized systems like Wordpress, even though I know that Publii could meet expectations perfectly.

Likewise, not all of us have the knowledge to be able to make a script to migrate our data in case of need

As for the help Publii needs, that's another matter, but I agree that everyone who uses Publii regularly should contribute financially, within their means. In my case I make a small monthly donation.

Thanks

**Reply by candidexmedia** [reply]
> But I have one fear, which has held me back from making more Publii-based websites: what would I do if one day the Publii project disappeared? 

Having some caution re: potential abandonware is good! Hopefully, the devs can chime in.

> ... or even have a tool to allow content to be exported to at least classic HTML for those people who want to switch to another system

@wefer-01 Doesn't Publii do this already with the `output` folder? Maybe I'm missing something...

> Because taking the render result directly and trying to edit it is a drama as it is really complex with scripts etc.

I think it would be tricky with any static site generator as well, since they all work in their own way and aren't necessarily interoperable. Perhaps the only way to ensure a site generator that you can use forever is to create your own? I, personally, don't have that kind of expertise or time to hand code all my pages, so I use this tool, and trust that there are enough users over here that someone will create something to save us all. 😂

And if not, I'll search or ask the AI overlords to help me create a script that takes the RSS feed and transforms it into the format I need.

> I agree that everyone who uses Publii regularly should contribute financially, within their means. In my case I make a small monthly donation.

Agreed! 💯

https://opencollective.com/publii

**Comment by bobmitro** [comment]
What a discussion :)

No worries at all; shutting down the project is definitely not on the horizon!  If, for some random reason, that were to happen, we would notify our community 1-2 years in advance.

Generally speaking, even if such a scenario were to occur, it wouldn't be the end of the world.  

Look at this: Most SaaS solutions typically stop working when they suspend their activities. Publii is a free program that works locally, so even after development stops, it will continue to function smoothly for a very long time. Plus, we build everything with standard practices, so there shouldn't be any sudden breaks.

The best part? Publii is open-source! That means anyone, even you, can jump in and help keep it going, even in some crazy future where we stop (which, again, is not happening!).

Publii is a passion project for us, and we've been pouring our hearts into it for eight years. We've got a big update coming out super soon, at the end of the month, with tons of cool new features. 
Here's a little sneak peek for the curious: check out the [Simple v.3 theme](https://demo.getpublii.eu/themes/simple/v3/), which includes some of the new upcoming features in Publii. I'm sure keen observers will notice these new features!

Oh, and about exporting your website data –  we hear you! We've been talking about adding that feature for a while now, and it's definitely on the table. It just needs a little time, but hopefully, we can make it happen this year.

So relax, keep building awesome websites. 
Thanks for being part of the Publii community!

**Reply by v7web** [reply]
Looks great, when is this new version of Simple theme likely to be released?

**Comment by johnwoltman** [comment]
Thanks for the reply, it's good to hear from the dev team on this :)

I've been checking out that Simple v3 theme and am excited to try it out when it's official!
